File: generate-favicon.py
#!/usr/bin/env python3
"""
Simple script to generate a favicon with "K" letter using PIL
Run with: python3 generate-favicon.py
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("⚠️  PIL (Pillow) not found.")
    print("\nTo generate the favicon automatically, install Pillow:")
    print("pip3 install Pillow")
    print("\nAlternatively:")
    print("1. Open create-favicon.html in your browser")
    print("2. Click 'Download favicon.ico'")
    print("3. Follow the instructions to convert PNG to ICO")
    exit(0)

# Create image
size = 32
img = Image.new('RGB', (size, size), color='#1890ff')
draw = ImageDraw.Draw(img)

# Try to use a good font, fallback to default
try:
    font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 24)
except:
    try:
        font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", 24)
    except:
        font = ImageFont.load_default()

# Draw the letter K in white
text = "K"
# Get text size for centering
bbox = draw.textbbox((0, 0), text, font=font)
text_width = bbox[2] - bbox[0]
text_height = bbox[3] - bbox[1]

# Center the text
x = (size - text_width) // 2
y = (size - text_height) // 2 - 2  # Slight adjustment for visual centering

draw.text((x, y), text, fill='white', font=font)

# Save as PNG first
png_path = './src/favicon-32x32.png'
img.save(png_path, 'PNG')
print(f'✅ Favicon PNG generated: {png_path}')

# Try to save as ICO
try:
    ico_path = './src/favicon.ico'
    img.save(ico_path, format='ICO', sizes=[(32, 32)])
    print(f'✅ Favicon ICO generated: {ico_path}')
    print('\n🎉 Success! Favicon is ready to use.')
except Exception as e:
    print(f'\n⚠️  Could not generate .ico file: {e}')
    print('\nTo convert PNG to ICO:')
    print('1. Visit: https://favicon.io/favicon-converter/')
    print('2. Upload src/favicon-32x32.png')
    print('3. Download and replace src/favicon.ico')

File: generate-favicon.js
// Simple script to generate a favicon with "K" letter
// Run with: node generate-favicon.js

const fs = require('fs');
const { createCanvas } = require('canvas');

// Check if canvas is installed
try {
    require.resolve('canvas');
} catch(e) {
    console.log('⚠️  Canvas module not found.');
    console.log('\nTo generate the favicon automatically, install canvas:');
    console.log('npm install canvas --save-dev');
    console.log('\nAlternatively, use the create-favicon.html file in your browser.');
    process.exit(0);
}

// Create 32x32 canvas for favicon
const size = 32;
const canvas = createCanvas(size, size);
const ctx = canvas.getContext('2d');

// Background - Blue (#1890ff)
ctx.fillStyle = '#1890ff';
ctx.fillRect(0, 0, size, size);

// Add slight rounded corners effect
ctx.globalCompositeOperation = 'destination-in';
ctx.beginPath();
ctx.moveTo(4, 0);
ctx.lineTo(28, 0);
ctx.quadraticCurveTo(32, 0, 32, 4);
ctx.lineTo(32, 28);
ctx.quadraticCurveTo(32, 32, 28, 32);
ctx.lineTo(4, 32);
ctx.quadraticCurveTo(0, 32, 0, 28);
ctx.lineTo(0, 4);
ctx.quadraticCurveTo(0, 0, 4, 0);
ctx.fill();

ctx.globalCompositeOperation = 'source-over';

// Letter K - White
ctx.fillStyle = '#ffffff';
ctx.font = 'bold 24px Arial, sans-serif';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';
ctx.fillText('K', 16, 17);

// Save as PNG
const pngBuffer = canvas.toBuffer('image/png');
fs.writeFileSync('./src/favicon-32x32.png', pngBuffer);

console.log('✅ Favicon PNG generated: src/favicon-32x32.png');
console.log('\nTo convert to .ico format:');
console.log('1. Visit: https://favicon.io/favicon-converter/');
console.log('2. Upload src/favicon-32x32.png');
console.log('3. Download and replace src/favicon.ico');
console.log('\nOr use an online tool or ImageMagick:');
console.log('convert src/favicon-32x32.png src/favicon.ico');
